### PR TITLE
use url set in S3_BACKEND as endpoint_url for s3 client 

### PR DESCRIPTION
**Pull request by alins**
In my scenario i would like to use localstack firehose with S3 somewhere else. 

First problem is that `aws_stack.get_s3_client` doesn't make use of `S3_BACKEND` at all. Second problem: my S3 requires `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` and this change will make use of them if the env variables are set

### CODE REVIEW

**Comment by alins** [comment]
in my testing i found something very strange, maybe i'm missing something. when i run this simple change in my IDE (PyCharm) everything checks out.

When i run the change in a docker container using a simple docker-compose file where i do set both `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY`...localstack is not using the values that i did set, instead is using the value `test` for both access key and secret key

further investigation reveals that *something* removes `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` environment variables..haven't figure out yet what doest that. How do i know they are removed? all my other env variables are there when i docker exec in the container and run a simple `printenv`, these two are not

File: localstack/utils/aws/aws_stack.py [hunk 1]
```diff
@@ -327,8 +327,13 @@ def fix_account_id_in_arns(response, colon_delimiter=':', existing=None, replace
 
 
 def get_s3_client():
+    if 'S3_BACKEND' in os.environ:
+        endpoint_url = os.environ.get('S3_BACKEND')
```
Comment (whummer) [inline localstack/utils/aws/aws_stack.py:331]: In certain cases, the environment variable `S3_BACKEND` could be set to an empty string (which would result in an error here). To make the code a bit more resilient, can we please change this to:
```
def get_s3_client():
    endpoint_url = os.environ.get('S3_BACKEND', '').strip() or config.TEST_S3_URL
    ...
```

**Comment by whummer** [comment]
Any updates on this one @alins - could you please take a look and provide an update / short response to the comments? Thanks!

**Comment by alins** [comment]
I will do my best to work on this today, i'm also curious to test with `latest` and see if this problem is actually fixed or not already

**Comment by whummer** [comment]
Hi @alins , thanks for your response. Did you have a chance to look into this one more time? Thanks for your support in resolving this!

**Comment by alins** [comment]
my apologies for such a long delay @whummer, this problem still happens with the latest version `0.11.4`. for context here's how a repro this problem (using [minio](https://github.com/minio/miniol) as external s3 service)

docker-compose.yml
```
version: '3.7'
services:
    localstack:
        image: localstack/localstack:0.11.4
        ports:
            - "4566:4566"
        environment:
            - SERVICES=firehose
            - DEFAULT_REGION=us-west-2
            - FORCE_NONINTERACTIVE=true
            - HOSTNAME=localhost
            - HOSTNAME_EXTERNAL=localhost
            - USER=root
            - LOCALSTACK_DURABLE_DATA=false
            - AWS_ACCESS_KEY=admin
            - AWS_SECRET_KEY=password
            - S3_BACKEND=http://minio:19180
            - DOCKER_HOST=unix:///var/run/docker.sock
        volumes:
            - "/tmp/localstack:/tmp/localstack"
            - "/var/run/docker.sock:/var/run/docker.sock"        
    minio:
        image: minio/minio
        ports:
            - "19180:19180"
        environment:
            - DEFAULT_REGION=us-west-2
            - MINIO_PORT=19180
            - MINIO_ACCESS_KEY=admin
            - MINIO_SECRET_KEY=password
```

so first `docker-compose up`...so far so good.

create firehose  (let's assume s3 bucket `testing-delivery-stream-bucket` already exists in minio)
```
--s3-destination-configuration '{"BucketARN": "arn:aws:s3:::'testing-delivery-stream-bucket'", "RoleARN": "bogus", "Prefix": "testing"}'
```

than run this python script to add data to firehose

```
import boto3
import json

# Create a boto client to use firehose service with specified endpoint
client = boto3.client("firehose",endpoint_url="http://localhost:4566")

# Just some data
data = {"name":"ruhshan", "attributes":["lazy", "crazy"]}

# Put record on the stream we created
client.put_record(Record={"Data":json.dumps(data)}, DeliveryStreamName="testing-delivery-stream")

````

at this point boto is throwing this exception
```
 botocore.exceptions.EndpointConnectionError: Could not connect to the endpoint URL: "http://localhost:0/testing-delivery-stream-bucket/testing/2020/09/02/21/testing-delivery-stream-2020-09-02-21-27-26-9f74e0f5-4176-4d28-9d3d-da40e6c165ba"
```

`localhost:0` doesn't look right

**Comment by whummer** [comment]
Thanks for clarifying @alins . We'll merge the PR as-is, and tackle the comment in a separate PR.

**Pull request merged**